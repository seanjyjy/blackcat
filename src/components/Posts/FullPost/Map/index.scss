@use 'src/styles/colors' as c;

@import 'src/styles/commonStyles';

:local {
  .name,
  .location {
    font-size: 14px;
    color: c.$primary-neutral;
  }

  .name {
    margin-top: 5px;
    font-weight: 600;
  }

  .map {
    margin-top: 10px;
    height: 94px;
    width: 98%;
    border-radius: 8px;
    transform: translateX(1%);
  }
}
