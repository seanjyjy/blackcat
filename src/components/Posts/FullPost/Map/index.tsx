import React from "react";

import LeftTitleContainer from "../../../LeftTitleContainer";
import MapImg from "../../../../images/gmap.png";

import styles from "./index.scss";

const Map = () => {
  return (
    <LeftTitleContainer header="Where">
      <>
        <p className={styles.name}>Marina Bay Sands</p>
        <p className={styles.location}>10 Bayfront Ave, S018956</p>
        <img src={MapImg} alt="map" className={styles.map} />
      </>
    </LeftTitleContainer>
  );
};

export default Map;
