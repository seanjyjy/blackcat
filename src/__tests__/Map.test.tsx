import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Map from "../components/Posts/FullPost/Map";

describe("snapshot testing", () => {
  it("snapshot for map", () => {
    const mapComponent = renderer.create(<Map />).toJSON();
    expect(mapComponent).toMatchSnapshot();
  });
});

describe("render logic", () => {
  it("render logic for map", () => {
    render(<Map />);

    expect(screen.getByText("Where")).toBeInTheDocument();
    expect(screen.getByText("Marina Bay Sands")).toBeInTheDocument();
    expect(screen.getByText("10 Bayfront Ave, S018956")).toBeInTheDocument();
  });
});
