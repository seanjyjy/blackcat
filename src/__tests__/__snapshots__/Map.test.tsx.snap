// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing snapshot for map 1`] = `
<div
  className="container"
>
  <div
    className="divider"
  />
  <div
    className="header"
  >
    <div />
    <p>
      Where
    </p>
  </div>
  <p
    className="name"
  >
    Marina Bay Sands
  </p>
  <p
    className="location"
  >
    10 Bayfront Ave, S018956
  </p>
  <img
    alt="map"
    className="map"
    src="test-file-stub"
  />
</div>
`;
